portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -11,41 +12,43 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.FetchOptions;
 import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
-import com.google.sps.objects.User;
+import com.google.sps.objects.Comment;
 
+/**
+ * A class to fetch data from the datastore library.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<User> userData = new ArrayList<User>();
+  /**@param userData The current data that has been fetched from the datastore
+   *     library, it acts as a local cache of the data that has been recorded.
+   *     Both the doGet and doPost methods rely on this local cache, only
+   *     refreshing if the refresh parameter is specified and true.
+   */
+  private ArrayList<Comment> userData = new ArrayList<Comment>();
   private DatastoreService storeData;
   // Intent is for the get method to be called on the load of the comment body.
   @Override
   public void init() {
     storeData = DatastoreServiceFactory.getDatastoreService();
   }
+  /**
+   * Get the value of the data stored in userData, only refresh this data from
+   * the database if refresh is defined and true in the request.
```

Comment:
I think you can refresh `userData` each time when you doPost, so doGet only need to read the `userData`.

Reply:
I was using the convention that the POST changes the user data on the server while the get merely fetches it. This is a bit ambiguous in the case where we fetch a certain number of entries from the datastore, but I thought it made more sense to handle this in a GET request than a POST one. In order to support this behavior both GET and POST requests must allow for a refresh of the data from the database.

Reply:
I think Jie's suggestion is more in line with that convention, right? If you keep the cache up to date in POST, you shouldn't need to modify data within GET. I'm not sure I understand what you mean by "This is a bit ambiguous in the case where we fetch a certain number of entries from the datastore."

Reply:
I was viewing modifying the cache and the database entries as 2 different things. So I was thinking as the refresh entry doesn’t modify the database it should be a get request even though it does modify the cache. I meant it was ambiguous in this case as you could also use the reasoning on the cache and think it’s a post request, but I was thinking that the spirit was more in whether or not the database itself was modified though maybe this isn’t the case?

Reply:
POST is the only operation that will modify data in datastore, to keep your cache consistent with datastore, you need to update your cache after writing to datastore, this can be done in POST. In your GET request, you just need to read from cache to avoid performance pressure on datastore, if it miss the hit(e.g. doesn't have enough entries in your case), it can update the cache by reading the enough latest data from datastore.
I think how to update the cache and datastore should be a inside logic instead of controlling by request parameters, users always want to get the latest data, they don't care about whether you use a cache or not and it is unnecessary to expose the refresh parameter to user

Reply:
Ok, that makes sense, I'll implement this change

Reply:
I think this change is big enough I'll make a new pull request.